docs/developer-docs/functionality/http_requests/http_requests-how-it-works.md

```diff
@@ -0,0 +1,152 @@
+
+The programming model of this feature is such that the canister making an HTTP outcall acts as *HTTP client*, i.e., needs to, in the general case, interpret various headers and act accordingly. For use cases such as API requests this is rather straightforward and does not require many specific considerations in the standard case. The IC protocol stack can conceptually be simply seen as a communication pipe between the canister and the conventional HTTP server that makes sure that the HTTP response makes it through consensus and all honest replicas receive the exact same response in execution, i.e., the calling canister receives an agreed-upon response.
+
+## Comparison with Oracles
+
+Blockchain oracles, or just oracles, are external parties that interact with the blockchain by making queries or sending ingress messages, just like any user. They have so far been the main means in the blockchain world of letting smart contracts communicate with traditional Web 2.0 servers.
+
+The basic architecture is that an oracle smart contract is deployed on chain. Users make requests to this oracle contract which stores them. An oracle, i.e., Web 2.0 server, obtains the stored request from the oracle smart contract through normal interaction with it using queries and update calls. Then the oracle executes the request in the Web 2.0 world by menans of a regular HTTP call and provides back the response to the oracle smart contract, again using the normal interaction paradigm with the blockchain (update call). Then the oracle smart contract provides back the response to the calling smart contract using a standard on-chain interaction, which concludes a brief introduction to how blockchain oracles work.
+
+In the oracle architecture, a single oracle may fulfill the original request, or a decentralized network of oracles may each issue the request and then provide an agreed-upon response (including evidence) to the oracle contract and ultimately to the calling canister. As we can see, the oracle world effectively can achieve the same result as HTTP outcalls, but with a more complex architecture of requiring one or more external parties that want to be paid for their services. Thus, one can assume that HTTP outcalls can replace oracles for many relevant use cases, and do so in a stronger trust model and with lower fees and lower request execution latency.
+
+Oracle services like ChainLink may provide additional functionality or services. One prominent example is using multiple data sources for a given information item, e.g., an asset price, and then providing a normalized response, e.g., the median price, as a result. Another example is keeping a selection of historic prices that have been retrieved available on chain for direct access by smart contracts without any oracle interaction. Both those functionalities can be built with the canister HTTP outcalls feature and do not require oracles.
```

Comment:
Is it a good idea to mention a specific provider by name?

Reply:
Maybe not, removed it.